cli/server.go

```diff
@@ -1309,6 +1309,30 @@ func templateHelpers(options *coderd.Options) map[string]any {
 	return map[string]any{
 		"base_url":     func() string { return options.AccessURL.String() },
 		"current_year": func() string { return strconv.Itoa(time.Now().Year()) },
+		"logo_url": func() string {
+			logoURL, err := options.Database.GetLogoURL(context.Background())
```

Comment:
Perhaps passing a context to `templateHelpers` (server lifetime) would be a good idea, and perhaps we should define a maximum execution time here (context.WithTimeout) so that we don't end up in a situation where the DB query simply blocks for an indefinite amount of time.

Reply:
👍 

I have similar feelings. We might be risking by placing DB call inside a template helper. I'm wondering if we shouldn't implement something similar to AppearanceFetcher for the site handler.

Reply:
Good catch 👍

Reply:
What would you say if we slightly modify this concept. My main concern is around swallowing the error in favor of returning an empty string `""`, so:

1. Wherever we render the `htmlBody`:

```go
	htmlBody, err = render.GoTemplate(htmlTemplate, payload, s.helpers)
	if err != nil {
		return nil, xerrors.Errorf("render full html template: %w", err)
	}
```

we create a dynamic `template.FuncMap` with `app_name` and `logo_url`. We could add a reference to the store to `SMTPHandler`, and retrieve values before rendering.

Thoughts?